diesel/src/query_builder/mod.rs

```diff
@@ -11,6 +11,7 @@ mod clause_macro;
 
 mod ast_pass;
 pub mod bind_collector;
+pub mod combination_clause;
```

Comment:
We do not want to expose that as public API. The necessary traits to constructs such queries should be exposed via `diesel::query_dsl::QueryDsl`. Additionally there should be helper type defs in `diesel::dsl` to refer to the constructed types.